docs/design/reserve-action.md

```diff
@@ -0,0 +1,261 @@
+# Reserve Action
+
+[@sivanzcw](https://github.com/sivanzcw); July 13, 2019
+
+## Table of Contents
+
+- [Reserve Action](#reserve-action)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Motivation](#motivation)
+  * [Function Detail](#function-detail)
+    + [Explanation of terms](#explanation-of-terms)
+      - [Job Ready](#job-ready)
+      - [Job Condition Ready](#job-condition-ready)
+      - [Job starving](#job-starving)
+      - [ForceAllocated](#forceallocated)
+      - [FakeAllocated](#fakeallocated)
+    + [Node resource reservation rule](#node-resource-reservation-rule)
+      - [Node reserve logic](#node-reserve-logic)
+      - [The proportion of nodes participating in reserve action](#the-proportion-of-nodes-participating-in-reserve-action)
+      - [Node selection strategy to participate in reserve](#node-selection-strategy-to-participate-in-reserve)
+      - [Node calculating score logic](#node-calculating-score-logic)
+  * [Data Model](#data-model)
+  
+## Introduction
+
+The `reserve` action is to solve the problem that the task can not be scheduled for a long time. We call these long 
```

Comment:
Could we solve it by `preempt`?

Reply:
if jobs/pods are in the same "priority", we can not trigger preemption :)